repos/spack_repo/builtin/packages/py_napari/package.py

```diff
@@ -0,0 +1,102 @@
+# Copyright Spack Project Developers. See COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+from spack_repo.builtin.build_systems.python import PythonPackage
+
+from spack.package import *
+
+
+class PyNapari(PythonPackage):
+    """napari is a fast, interactive, multi-dimensional image viewer for
+    Python. It's designed for browsing, annotating, and analyzing large
+    multi-dimensional images. It's built on top of Qt (for the GUI), vispy (for
+    performant GPU-based rendering), and the scientific Python stack (numpy,
+    scipy)."""
+
+    homepage = "https://www.napari.org"
+    pypi = "napari/napari-0.6.5.tar.gz"
+
+    maintainers("Markus92")
+
+    license("BSD-3-Clause", checked_by="Markus92")
+
+    version("0.6.5", sha256="00663221583467f41fa956a672b0b4cd7a761e2642db41fc3c336d1c82339533")
+    version("0.4.18", sha256="daea9ab94124140fc0f715e945dd1dd6dc3056a1cb2f2cc31fc29b80162943e4")
+
+    variant("all", default=False, description="Install all optional dependencies")
+
+    depends_on("python@3.10:", type=("build", "run"))
+
+    depends_on("py-setuptools@69:", when="@0.6:", type="build")
+    depends_on("py-setuptools@56:", type="build")
+    depends_on("py-setuptools-scm@8: +toml", when="@0.6:", type="build")
+    depends_on("py-setuptools-scm@3.4: +toml", type="build")
+
+    depends_on("py-appdirs@1.4.4:", type=("build", "run"))
+    depends_on("py-app-model@0.4.0:0.4", when="@0.6:", type=("build", "run"))
+    depends_on("py-app-model@0.1:0.2", when="@0.4:0.5", type=("build", "run"))
+    depends_on("py-cachey@0.2.1:", type=("build", "run"))
+    depends_on("py-certifi@2018.1.18:", type=("build", "run"))
+    depends_on("py-dask@2021.10.0: +array", type=("build", "run"))
+    depends_on("py-imageio@2.20:", type=("build", "run"))
+    conflicts("^py-imageio@2.28.0")
```

Comment:
[minor] this can be simplified
```suggestion
    depends_on("py-imageio@2.20:2.27,2.28.1:", type=("build", "run"))
```